src/Http/Routing/src/EndpointMiddleware.cs

```diff
@@ -52,6 +60,34 @@ public async Task Invoke(HttpContext httpContext)
             await _next(httpContext);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void EnsureRequisiteMiddlewares(HttpContext httpContext, Endpoint endpoint)
+        {
+            if (endpoint.Metadata.GetMetadata<IAuthorizeData>() != null &&
+                !httpContext.Items.ContainsKey(AuthorizationMiddlewareInvokedKey))
+            {
+                ThrowMissingAuthMiddlewareException(endpoint);
+            }
+
+            if (endpoint.Metadata.GetMetadata<ICorsMetadata>() != null &&
+                !httpContext.Items.ContainsKey(CorsMiddlewareInvokedKey))
+            {
+                ThrowMissingCorsMiddlewareException(endpoint);
+            }
+        }
+
+        private static void ThrowMissingAuthMiddlewareException(Endpoint endpoint)
+        {
+            throw new InvalidOperationException($"Endpoint {endpoint.DisplayName} contains authorization metadata, " +
```

Comment:
This needs to include instructions about how to fix the problem. Imagine you see this error for the first time in preview 5.